sonar-scanner-engine/src/main/java/org/sonar/scanner/task/ScanTask.java

```diff
@@ -44,6 +45,8 @@ public ScanTask(TaskContainer taskContainer, TaskProperties taskProps) {
   @Override
   public void execute() {
     AnalysisProperties props = new AnalysisProperties(taskProps.properties(), taskProps.property(CoreProperties.ENCRYPTION_SECRET_KEY_PATH));
-    new ProjectScanContainer(taskContainer, props).execute();
+    ProjectScanContainer scanContainer = new ProjectScanContainer(taskContainer, props);
+    scanContainer.add(DefaultAnalysisMode.class);
+    scanContainer.execute();
```

Comment:
Is there a reason? I think it's better to have all classes added to the container in one place.